lib/rex/proto/mysql/client.rb

```diff
@@ -28,6 +28,32 @@ def current_database
           # Current database is stored as an array under the type 1 key.
           session_track.fetch(1, ['']).first
         end
+
+        # @return [Hash] Server version variables:
+        #  * :arch [String] The server architecture.
+        #  * :platform [String] The server platform.
+        #  * :hostname [String] The server hostname.
+        #  * :datadir [String] The server data directory.
+        #  * :version [String] The server version. This is the same as calling #server_info on the MySQL client.
+        def query_server_vars
+          result = {}
+
+          server_vars = query('show variables').entries
```

Comment:
Would it make more sense to be specific here with what we want to query 👀 

```
mysql @ 127.0.0.1:3306 > query "show variables where variable_name in ('version_compile_machine', 'version_compile_os')"
Response
========

    #  Variable_name            Value
    -  -------------            -----
    0  version_compile_machine  x86_64
    1  version_compile_os       Linux

```

Disclaimer: I think the result will be unordered - so you'll maybe want a `sort` call 

Or maybe:

``
mysql @ 127.0.0.1:3306 > query "SELECT @@GLOBAL.version_compile_machine, @@GLOBAL.version_compile_os;"
Response
========

    #  @@GLOBAL.version_compile_machine  @@GLOBAL.version_compile_os
    -  --------------------------------  ---------------------------
    0  x86_64                            Linux

```